gcp_variant_transforms/pipeline_common.py

```diff
@@ -66,82 +66,68 @@ def parse_args(argv, command_line_options):
     return PipelineModes.LARGE
 
-  if input_pattern is not None:
-    total_files = _get_pipeline_mode_for_pattern_input(input_pattern)
-  else:
-    total_files = _get_pipeline_mode_for_file_input(input_file)
+  match_results = filesystems.FileSystems.match(input_patterns)
+  if not match_results:
+    if input_file:
+      raise ValueError(
+          'No files matched any input pattern in {} file'.format(input_file))
+    else:
+      raise ValueError(
+          'No files matched {} input pattern'.format(input_patterns[0]))
+
+  total_files = 0
+  for match in match_results:
+    if not match.metadata_list:
+      raise ValueError(
+          'Input pattern {} did not match any files.'.format(match.pattern))
+    total_files += len(match.metadata_list)
 
   if total_files > _LARGE_DATA_THRESHOLD:
     return PipelineModes.LARGE
   elif total_files > _SMALL_DATA_THRESHOLD:
     return PipelineModes.MEDIUM
   return PipelineModes.SMALL
 
-def _get_pipeline_mode_for_pattern_input(input_pattern):
-  match_results = filesystems.FileSystems.match([input_pattern])
-  if not match_results:
-    raise ValueError('No files matched input_pattern: {}'.format(input_pattern))
-  return len(match_results[0].metadata_list)
-
-def _get_pipeline_mode_for_file_input(input_file):
-  if not filesystems.FileSystems.exists(input_file):
-    raise ValueError('Input file {} doesn''t exist'.format(input_file))
-  all_patterns = []
-  with filesystems.FileSystems.open(input_file) as f:
-    for _, l in enumerate(f):
-      all_patterns.append(l.strip())
-    if not all_patterns:
-      raise ValueError('Input file {} is empty.'.format(input_file))
-  match_results = filesystems.FileSystems.match(all_patterns)
-  if not match_results:
-    raise ValueError(
-        'No files matched any input pattern in {} file'.format(input_file))
-  total_number = 0
-  for match in match_results:
-    if not match.metadata_list:
-      raise ValueError(
-          'Input pattern {} from {} did not match any files.'.format(
-              match.pattern, input_file))
-    total_number += len(match.metadata_list)
-  return total_number
-
-def get_file_names(input_file):
-  #if (input_pattern):
-  #  return [input_pattern]
-  result = []
-  if not filesystems.FileSystems.exists(input_file):
-    raise ValueError('Input file {} doesn''t exist'.format(input_file))
-  with filesystems.FileSystems.open(input_file) as f:
-    for _, l in enumerate(f):
-      result.append(l)
-    if not result:
-      raise ValueError('Input file {} is empty.'.format(input_file))
-    return result
-
 
-def read_headers(pipeline, pipeline_mode, input_pattern, input_file):
+def read_headers(pipeline, pipeline_mode, input_patterns):
   # type: (beam.Pipeline, int, str) -> pvalue.PCollection
   """Creates an initial PCollection by reading the VCF file headers."""
-  if pipeline_mode == PipelineModes.LARGE or input_file:
+  if pipeline_mode == PipelineModes.LARGE or len(input_patterns) > 1:
```

Comment:
It might be easier to just change the pipeline_mode to LARGE in `get_pipeline_mode` when we have more than one input_pattern. But please check whether we use the pipeline mode anywhere else rather than reading the variants.

Reply:
Alas, annotation also looks up the pipeline mode, while it can have different input pattern (specifically, single input pattern of the shard directory).

Reply:
The annotation pipeline also uses the pipeline_mode to read variants, I don't think it is different from other cases, right?

Reply:
I'm sorry, I wasn't clear - at this [step](https://github.com/googlegenomics/gcp-variant-transforms/blob/master/gcp_variant_transforms/pipeline_common.py#L105) we potentially already have sharded the output so only have 1 input pattern, but whether input was large or not is still relevant